### ISSUE TITLE
Failed to add 1.9.0 to ios platform - exit code null for pod install

### ISSUE THREAD

**Issue by yosiat** [phonegap/phonegap-plugin-push#1367]
### Expected Behaviour
succeed to install 1.9.0 to ios platform

### Actual Behaviour

```
No scripts found for hook "before_plugin_add".
No version specified for phonegap-plugin-push, retrieving version from config.xml
No version for phonegap-plugin-push saved in config.xml
Attempting to use npm info for phonegap-plugin-push to choose a compatible release
Running command: /Users/yosi/code/yosi-app/platforms/android/cordova/version
Running command: /Users/yosi/code/yosi-app/platforms/browser/cordova/version
Running command: /Users/yosi/code/yosi-app/platforms/ios/cordova/version
Command finished with error code 0: /Users/yosi/code/yosi-app/platforms/android/cordova/version
Command finished with error code 0: /Users/yosi/code/yosi-app/platforms/browser/cordova/version
Command finished with error code 0: /Users/yosi/code/yosi-app/platforms/ios/cordova/version
Ignoring invalid version in phonegap-plugin-push cordovaDependencies: 2.0.0 (must be a single version <= latest or an upper bound)
Calling plugman.fetch on plugin "phonegap-plugin-push@1.9.0"
Calling plugman.install on plugin "/Users/yosi/code/yosi-app/plugins/phonegap-plugin-push" for platform "android
Installing "phonegap-plugin-push" for android
Running command: /Users/yosi/code/yosi-app/platforms/android/cordova/version
Command finished with error code 0: /Users/yosi/code/yosi-app/platforms/android/cordova/version
Finding scripts for "before_plugin_install" hook from plugin phonegap-plugin-push on android platform only.
No scripts found for hook "before_plugin_install".
Install start for "phonegap-plugin-push" on android.
ANDROID_HOME=/Users/yosi/Library/Android/sdk
JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_112.jdk/Contents/Home
Subproject Path: CordovaLib
Running command: /Users/yosi/code/yosi-app/platforms/android/gradlew clean -b /Users/yosi/code/yosi-app/platforms/android/build.gradle -Dorg.gradle.daemon=true -Dorg.gradle.jvmargs=-Xmx2048m -Pandroid.useDeprecatedNdk=true
Parallel execution is an incubating feature.
Incremental java compilation is an incubating feature.
:clean
:CordovaLib:clean

BUILD SUCCESSFUL

Total time: 1.039 secs
Command finished with error code 0: /Users/yosi/code/yosi-app/platforms/android/gradlew clean,-b,/Users/yosi/code/yosi-app/platforms/android/build.gradle,-Dorg.gradle.daemon=true,-Dorg.gradle.jvmargs=-Xmx2048m,-Pandroid.useDeprecatedNdk=true
Beginning processing of action stack for android project...
Installing Android library: push.gradle
Installing Android library: com.android.support:support-v13:23+
Installing Android library: com.google.android.gms:play-services-gcm:9.8+
Installing Android library: me.leolin:ShortcutBadger:1.1.4@aar
Action stack processing complete.
Updating build files since android plugin contained <framework>
Subproject Path: CordovaLib
Install complete for phonegap-plugin-push on android.
Finding scripts for "after_plugin_install" hook from plugin phonegap-plugin-push on android platform only.
No scripts found for hook "after_plugin_install".
Calling plugman.install on plugin "/Users/yosi/code/yosi-app/plugins/phonegap-plugin-push" for platform "browser
Installing "phonegap-plugin-push" for browser
Running command: /Users/yosi/code/yosi-app/platforms/browser/cordova/version
Command finished with error code 0: /Users/yosi/code/yosi-app/platforms/browser/cordova/version
Finding scripts for "before_plugin_install" hook from plugin phonegap-plugin-push on browser platform only.
No scripts found for hook "before_plugin_install".
Install start for "phonegap-plugin-push" on browser.
Error loading cordova-browser
Beginning processing of action stack for browser project...
Action stack processing complete.
Install complete for phonegap-plugin-push on browser.
Finding scripts for "after_plugin_install" hook from plugin phonegap-plugin-push on browser platform only.
No scripts found for hook "after_plugin_install".
Calling plugman.install on plugin "/Users/yosi/code/yosi-app/plugins/phonegap-plugin-push" for platform "ios
Installing "phonegap-plugin-push" for ios
Running command: /Users/yosi/code/yosi-app/platforms/ios/cordova/version
Command finished with error code 0: /Users/yosi/code/yosi-app/platforms/ios/cordova/version
Finding scripts for "before_plugin_install" hook from plugin phonegap-plugin-push on ios platform only.
No scripts found for hook "before_plugin_install".
Install start for "phonegap-plugin-push" on ios.
Beginning processing of action stack for ios project...
Action stack processing complete.
Adding pods since the plugin contained <framework>(s) with type="podspec"
pods.json found in platforms/ios
Podfile found in platforms/ios
Set pods.json for `GoogleCloudMessaging`
Added pod line for `GoogleCloudMessaging`
Set pods.json for `GGLInstanceID`
Added pod line for `GGLInstanceID`
Wrote to pods.json.
Wrote to Podfile.
Running `pod install` (to install plugins)
Running command: pod install --verbose
==== pod install start ====

  Preparing


Analyzing dependencies

Inspecting targets to integrate
  Using `ARCHS` setting to build architectures of target `Pods-Bringg Dispatcher`: (``)

Resolving dependencies of `Podfile`

Comparing resolved specification to the sandbox manifest
  A GGLInstanceID
  A GoogleCloudMessaging
  A GoogleIPhoneUtilities
  A GoogleInterchangeUtilities
  A GoogleSymbolUtilities
  A GoogleUtilities

Downloading dependencies

-> Installing GGLInstanceID (1.2.1)
  > Copying GGLInstanceID from `/Users/yosi/Library/Caches/CocoaPods/Pods/Release/GGLInstanceID/1.2.1-4a317` to `Pods/GGLInstanceID`

-> Installing GoogleCloudMessaging (1.2.0)
  > Copying GoogleCloudMessaging from `/Users/yosi/Library/Caches/CocoaPods/Pods/Release/GoogleCloudMessaging/1.2.0-f37ea` to `Pods/GoogleCloudMessaging`

-> Installing GoogleIPhoneUtilities (1.2.1)
  > Copying GoogleIPhoneUtilities from `/Users/yosi/Library/Caches/CocoaPods/Pods/Release/GoogleIPhoneUtilities/1.2.1-63f25` to `Pods/GoogleIPhoneUtilities`

-> Installing GoogleInterchangeUtilities (1.2.2)
  > Copying GoogleInterchangeUtilities from `/Users/yosi/Library/Caches/CocoaPods/Pods/Release/GoogleInterchangeUtilities/1.2.2-d5bc4` to `Pods/GoogleInterchangeUtilities`

-> Installing GoogleSymbolUtilities (1.1.2)
  > Copying GoogleSymbolUtilities from `/Users/yosi/Library/Caches/CocoaPods/Pods/Release/GoogleSymbolUtilities/1.1.2-631ee` to `Pods/GoogleSymbolUtilities`

-> Installing GoogleUtilities (1.3.2)
  > Copying GoogleUtilities from `/Users/yosi/Library/Caches/CocoaPods/Pods/Release/GoogleUtilities/1.3.2-8bbc7` to `Pods/GoogleUtilities`
  - Running pre install hooks

Generating Pods project
  - Creating Pods project
  - Adding source files to Pods project
  - Adding frameworks to Pods project
  - Adding libraries to Pods project
  - Adding resources to Pods project
  - Linking headers
  - Installing targets
    - Installing target `Pods-Bringg Dispatcher` iOS 8.0

  - Running post install hooks
  - Writing Xcode project file to `Pods/Pods.xcodeproj`
    - Generating deterministic UUIDs

Command finished with error code null: pod install,--verbose
Failed to install 'phonegap-plugin-push':Error: pod: Command failed with exit code null
    at ChildProcess.whenDone (/Users/yosi/code/yosi-app/platforms/ios/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:827:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
Error: pod: Command failed with exit code null
    at ChildProcess.whenDone (/Users/yosi/code/yosi-app/platforms/ios/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:827:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
```

### Reproduce Scenario (including but not limited to)

- cordova plugin rm phonegap-plugin-push
- cordova plugin add phonegap-plugin-push --variable SENDER_ID="..." --verbose

#### Cordova CLI version and cordova platform version

    cordova --version    6.4.0
    cordova platform version android      6.00
    cordova platform version ios      4.3.0
    pod --version  1.1.1
    macOS sierra 10.12.1

#### Plugin version

    Latest 1.9.0

**Comment by yosiat** [comment]
I know this issue looks the same as: https://github.com/phonegap/phonegap-plugin-push/issues/1342

But I am getting exit code "null" on what whom seems like successful pod install.

**Comment by macdonst** [comment]
@yosiat what version of CocoaPods do you have installed?

@shazron have you ever seen anything like this?

**Comment by yosiat** [comment]
@macdonst 1.1.1 
```
cordova --version    6.4.0
cordova platform version android      6.00
cordova platform version ios      4.3.0
pod --version  1.1.1
macOS sierra 10.12.1
```

**Comment by shazron** [comment]
At the time of implementation and testing, we tested with CocoaPods 1.0.1 because that was the latest released and stable version. Perhaps something changed with the newer version, and we need to re-test.

**Comment by shazron** [comment]
@yosiat not sure if this is a big hassle, but see if you can downgrade to CocoaPods 1.0.1 and re-test. If that succeeds, that isolates the problem to CocoaPods 1.1.1

**Comment by yosiat** [comment]
@shazron It works with cocoapods 1.0.1

**Comment by macdonst** [comment]
@shazron Odd, cause I was using 1.1.1

**Comment by shazron** [comment]
Just re-tested using this procedure, it succeeded (although there was a bunch of curl output to stdout for downloads that I didn't see before):
```
cordova create test test test
cd test
cordova platform add ios
cordova plugin add phonegap-plugin-push
cordova build
```

Versions:
```
cordova --version    6.4.0
cordova platform version ios      4.3.0
pod --version  1.1.1
macOS sierra 10.12.1 (16B2555)
```

This was the output I saw that I mentioned:
```
$ cordova plugin add phonegap-plugin-push
Fetching plugin "phonegap-plugin-push@1.9.0" via npm
Installing "phonegap-plugin-push" for ios
     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                    Dload  Upload   Total   Spent    Left  Speed

     0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
   100 1085k    0 1085k    0     0  1666k      0 --:--:-- --:--:-- --:--:-- 1673k

     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                    Dload  Upload   Total   Spent    Left  Speed

     0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
   100 1596k    0 1596k    0     0  3119k      0 --:--:-- --:--:-- --:--:-- 3124k

     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed

   100  119k  100  119k    0     0   357k      0 --:--:-- --:--:-- --:--:--  359k

     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed

     0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
   100 1939k    0 1939k    0     0  3745k      0 --:--:-- --:--:-- --:--:-- 3751k

     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

                                    Dload  Upload   Total   Spent    Left  Speed

     0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
   100 12789  100 12789    0     0  58099      0 --:--:-- --:--:-- --:--:-- 58397

     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                    Dload  Upload   Total   Spent    Left  Speed

     0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
   100 1207k    0 1207k    0     0  2196k      0 --:--:-- --:--:-- --:--:-- 2198k
```

I upgraded cocoapods from 1.0.1 to 1.1.1 using:
```
gem update cocoapods
```

**Comment by shazron** [comment]
Also, my ruby is: 
```
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]
```

Other info:
```
$ gem update cocoapods
Updating installed gems
Updating cocoapods
Fetching: cocoapods-core-1.1.1.gem (100%)
Successfully installed cocoapods-core-1.1.1
Fetching: claide-1.0.1.gem (100%)
Successfully installed claide-1.0.1
Fetching: cocoapods-deintegrate-1.0.1.gem (100%)
Successfully installed cocoapods-deintegrate-1.0.1
Fetching: cocoapods-downloader-1.1.2.gem (100%)
Successfully installed cocoapods-downloader-1.1.2
Fetching: cocoapods-trunk-1.1.1.gem (100%)
Successfully installed cocoapods-trunk-1.1.1
Fetching: molinillo-0.5.4.gem (100%)
Successfully installed molinillo-0.5.4
Fetching: CFPropertyList-2.3.3.gem (100%)
Successfully installed CFPropertyList-2.3.3
Fetching: nanaimo-0.2.2.gem (100%)
Successfully installed nanaimo-0.2.2
Fetching: xcodeproj-1.4.1.gem (100%)
Successfully installed xcodeproj-1.4.1
Fetching: fourflusher-2.0.1.gem (100%)
Successfully installed fourflusher-2.0.1
Fetching: gh_inspector-1.0.2.gem (100%)
Successfully installed gh_inspector-1.0.2
Fetching: cocoapods-1.1.1.gem (100%)
Successfully installed cocoapods-1.1.1
[removed a bunch of doc install logs...]
Gems updated: CFPropertyList claide cocoapods cocoapods-core cocoapods-deintegrate cocoapods-downloader cocoapods-trunk fourflusher gh_inspector molinillo nanaimo xcodeproj
```

It's possible that some gems have been updated in the interval between when you installed and I installed 1.1.1, possibly fixing some bugs since some gemspecs have ranges: https://rubygems.org/gems/cocoapods/versions/1.1.1

Try uninstalling and re-installing the cocoapods gem perhaps to see if this is the case.

**Comment by shazron** [comment]
I didn't go too deep into the dependency rabbit hole, but `xcodeproj` had two updates since 1.1.1 release: https://rubygems.org/gems/xcodeproj

**Comment by macdonst** [comment]
@shazron using:

```
$ pod --version
1.1.1
$ xcodeproj --version 
1.4.1
```

and I can't reproduce.

**Comment by yosiat** [comment]
My ruby version is:
```
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin16]
```

And I install cocoapods 1.1.1 from homebrew `brew install cocoapods`

**Comment by shazron** [comment]
@yosiat did you try reinstalling cocoapods? My point above was that a dependency was updated that you may not have picked up from an older install.

**Comment by Kjens93** [comment]
`cordova platform remove ios` and `cordova prepare ios` solved this for me.

**Comment by yosiat** [comment]
@shazron yes.

**Comment by jcesarmobile** [comment]
I used 1.1.1 too and worked fine.

Can you install with `gem install cocoapods` instead of using homebrew? (might need sudo)

**Comment by shazron** [comment]
@Kjens93 should there be a `cordova platform add ios` in the middle there after remove? Or I'm assuming you had the platform saved already in config.xml most likely thus `prepare` restored it.

**Comment by tgensol** [comment]
I got the same issue, impossible for me to install the plugin on ios :

`
[!] The Podfile does not contain any dependencies.


Error: pod: Command failed with exit code 1 Error output:
[!] The Podfile does not contain any dependencies.
`

and then 

`Failed to install 'phonegap-plugin-push':Error: pod: Command failed with exit code 1`

when I run ionic build ios I got this message : 

`Cordova/build-debug.xcconfig' because it could not be loaded.
	Reason: File could not be parsed due to preprocessing errors:
    pods-debug.xcconfig line 2: Unable to find included file "Pods/Target Support Files/Pods-X/Pods-X.debug.xcconfig"`

Hope this help

**Comment by macdonst** [comment]
@tgensol what version of cordova-ios are you using?

**Comment by Lestt** [comment]
Hi guys,

I have same kind of issue with this plugin on iOs(no problem with android). I try to compile my project and i got this error:

```
Linker command failed with exit code 1 (use -v to see invocation)
ld: library not found for -Pods-MyProject
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I am a newbie in iOs environment and mac, so i don't what information do you need.
Cocoapods version is **1.1.1** (installed via `gem install`), cordova-ios version is **4.3.0**.
I tried this command (i'm using ionic):

```
ionic platform rm ios
ionic state reset  --plugins
ionic platform add ios
cd /platforms/ios
pod install
```

And then open project in xcode and try to build and run it.
Sry if i'm not in the good thread, feel free to move it if needed (or ask me to create a new one).

**Comment by Lestt** [comment]
Ok i was opening the .xcodeproj in xcode instead of the xcworkspace. Now the build is ok. Sry for the spam.

**Comment by tgensol** [comment]
@macdonst  I have : 

cordova 6.4.0
android 6.0.0
ios 4.3.0
pod 1.1.1

**Comment by tgensol** [comment]
I tried with a new project, I got this again 

`Failed to install 'phonegap-plugin-push':Error: pod: Command failed with exit code 1
    at ChildProcess.whenDone (path_removed/plugin/platforms/ios/cordova/node_modules/cordova-common/src/superspawn.js:169:23)
    at emitTwo (events.js:106:13)
`

**Comment by macdonst** [comment]
Anyone getting the error:

```
Installing "phonegap-plugin-push" for ios
Failed to install 'phonegap-plugin-push':Error: pod: Command failed with exit code 1
```

should go to the command line and run `pod repo update` then try to re-install the plugin.

**Comment by tgensol** [comment]
Your command line was not working, I removed cocoapods with 'gem uninstall cocoapods', and then used homebrew to install cocoapods, and it is working now !

**Comment by shazron** [comment]
An update. After updating Ruby and stuff to get Jekyll working for another project on the same system, it somehow "corrupted" or upgraded some of my Ruby gems (fun!) and I got the same `Command failed with exit code null` error.

I uninstalled and reinstalled cocoapods (current release, not any pre-release) using the steps [here](http://stackoverflow.com/a/40828560/219684) then it all works again. Basically what others in this thread have found as well...
